NitroxClient/GameLogic/ServerTime.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using NitroxClient.Communication;
+using NitroxModel.Logger;
+
+namespace NitroxClient.GameLogic
+{
+    /// <summary>
+    /// The synchronized UTC server time.
+    /// </summary>
+    public class ServerTime
```

Comment:
Seems like a lot of effort to sync random numbers.  Was there not an option to make the random number deterministic instead?

Reply:
I had a discussion with @RabidCrab about this in discord; turns out this was only used for prediction, something which is completely irrelevant in this case. It was also mentioned that we need the simulation ownership framework to decide which client is authorative on the fires and damage numbers; as far as I understand this is to prevent clients spawning fires at different positions, only to be removed again once the server validated the damage/fire state array of one of the clients.

It wouldn't make sense to synchronize the random number generator anyway, since the entire codebase uses it. Once the clients call it a different amount of times it's already out of sync. A much better solution would be to rng the numbers on the simulating/owning client, and distribute that to the others (where the random generation of fires/damages is blocked, obviously).

Reply:
Ok, so we will forego this logic in favor of updates to simulation framework?

Reply:
I dumped this idea after talking with @MarijnS95 about it and coming to the conclusion that it's not necessary to synchronize the damage and fire if there's a designated owner. At the moment there's no ownership designation, but @MarijnS95 is working on a system that I'll be using when it's ready.

I'll remove all of the time syncing code, unless we want to keep it for a day/night cycle accurate down to the milliseconds. Considering the game code rounds time to the nearest second, I doubt it'll be needed.

Reply:
IIrc the time is currently only synced once with the server, when the client joins. Over time this may get skewed, or command might be used to alter the time. In any case, we should reuse or improve the existing system (if I read correctly, this PR adds a separate method of syncing time).

Reply:
@MarijnS95 , it uses the existing time system, but extends it to include the server's UTC time. When the client receives a TimeChange packet without a ClientSentAtMilliseconds (happens after authentication), it sets that value to the current time in milliseconds and sends the packet back to the server. The server will take the returned packet, set CurrentServerUtcTimeMilliseconds, and send the packet back to the client. The round-trip time is divided by 2 and added to the server time passed with the packet to account for latency.

Time is then determined by adding the current time delta to that received time. As long as their clock doesn't stop working, the time will be accurate with the server, generally within a few milliseconds.

Although the solution works, I can't think of any reason why we would need to know the server's time down to the milliseconds beyond what I originally intended for it to do.